Updating repo guidelines

zjalexander opened the pull request:
1. Clarify naming, especially around "x".
2. Clarify the primary mantainer role is to act on PRs
3. Removed suggestions around submitting new resource modules to this repo. 

Of these, 1. and 2. should be pretty straightforward.
The final item is reflecting reality - we haven't been able to take any new resource modules, and we need to be honest with ourselves about it :)

Going to keep this open for discussion for a while.
cc: @johlju @grayzu @mgreenegit @BrianFarnhill @PlagueHO @tysonjhayes @RichardSiddaway @raandree @mhendric @bgelens (phew!!!) 
Helps address #307 and some items brought up in the last community call. 


Plan from here on out:
1. Get consensus on above
2. Clarify "community supported" process
3. Start dealing with #307

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/powershell/dscresources/317)
<!-- Reviewable:end -->

KosherSalt commented on CONTRIBUTING.md:
```diff
@@ -364,23 +362,15 @@ All mof-based resource (with Get/Set/Test-TargetResource and a schema.mof file)
 ## Submitting a New Resource Module
-If you would like a new DSC resource module added to the DSC Resource Kit, please check the [existing issues](https://github.com/PowerShell/DscResources/issues) in this repository first to make sure no one else is already working on a similar module.
-If someone is already working on a similar module, please leave a comment on the open issue or add a GitHub reaction to the top comment to **express your interest**. You can also offer help and use the issue to coordinate your efforts.
+This repository is **not** accepting new modules at this time. We reccomend authoring the resource in your own repository and [submitting it to the gallery](https://docs.microsoft.com/en-us/powershell/gallery/psgallery/creating-and-publishing-an-item) under your own name. However, feel free to bring up any modules you have authored during the [DSC Resoucrce Kit Community Call](https://github.com/ ...
```
Change reccomend to recommend

johlju commented:
Reviewed 2 of 3 files at r1, 1 of 1 files at r2.
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---

*[CONTRIBUTING.md, line 367 at r2](https://reviewable.io:443/reviews/powershell/dscresources/317#-Ksigf_CbQN0VBVsnr-g:-Ksigf_CbQN0VBVsnr-h:b-yqoy2d) ([raw file](https://github.com/powershell/dscresources/blob/4b7a21c78536bf6af98ffa6667dca79d91646754/CONTRIBUTING.md#L367)):*
> ```Markdown
> If the module name does not start with 'x', the resource name should not start with 'x' (e.g. MSFT_Resource.psm1 or Resource.psm1).
> 
> AAny test or example files for the resource should be named to match the files for the same resource. For example, if the main resource file is named 'MSFT_xResource.psm1', then the unit test file should be named 'MSFT_xResource.Tests.ps1'. Consistent naming helps the review process. 
> ```

Typo 'Any'

---

*[CONTRIBUTING.md, line 397 at r2](https://reviewable.io:443/reviews/powershell/dscresources/317#-KsihwgRtN2YhPLrVcUu:-KsihwgRtN2YhPLrVcUv:b-xtgaon) ([raw file](https://github.com/powershell/dscresources/blob/4b7a21c78536bf6af98ffa6667dca79d91646754/CONTRIBUTING.md#L397)):*
> All new resource modules should

Maybe "All new resources in existing resource modules should..."  if you read the changed header.

If not, then this section should be revised. Naming can not be relevant if  users should own there resource modules.
Or could these resources be linked to DscResourceKit as Community Supported HQRM?

---

*[Maintainers.md, line 24 at r2](https://reviewable.io:443/reviews/powershell/dscresources/317#-KsiixuN6tqJ4lqqWh1N:-KsiixuN6tqJ4lqqWh1O:b-potrft) ([raw file](https://github.com/powershell/dscresources/blob/4b7a21c78536bf6af98ffa6667dca79d91646754/Maintainers.md#L24)):*
> up-to-date on  what

Two spaces between 'on' and 'what'.

---

*[Maintainers.md, line 24 at r2](https://reviewable.io:443/reviews/powershell/dscresources/317#-Ksij4lNoN8JZucFHT6j:-Ksij4lOdGHw2im5LA07:b-d40t14) ([raw file](https://github.com/powershell/dscresources/blob/4b7a21c78536bf6af98ffa6667dca79d91646754/Maintainers.md#L24)):*
> repos

Change to repositories?

---

*[Maintainers.md, line 96 at r2](https://reviewable.io:443/reviews/powershell/dscresources/317#-Ksij9LRORxb0mIw_BlM:-Ksij9LRORxb0mIw_BlN:b77rr09) ([raw file](https://github.com/powershell/dscresources/blob/4b7a21c78536bf6af98ffa6667dca79d91646754/Maintainers.md#L96)):*
> reqeusts

Typo. And add full stop at the end.

---

*[Naming.md, line 12 at r2](https://reviewable.io:443/reviews/powershell/dscresources/317#-Ksik-zSA8lPhCxdIpSs:-Ksik-zSA8lPhCxdIpSt:bvuefnt) ([raw file](https://github.com/powershell/dscresources/blob/4b7a21c78536bf6af98ffa6667dca79d91646754/Naming.md#L12)):*
> ```Markdown
> The "x" prefix means a resource is "experimental". It comes with no support guarantee whatsoever.
> 
> ## Community Supported
> ```

Is this a resource modules already present in the DSC Resource Kit only?  

When is it okay to rename a module from "x" to "Dsc"?  Community approval process? For example renaming xSQLServer. A community developed by a contributor outside of DSC Resource Kit we could only recommend not name to "Dsc" unless it's HQRM.

---

*[Naming.md, line 13 at r2](https://reviewable.io:443/reviews/powershell/dscresources/317#-KsijQahTxK12rXHPKJk:-KsijQahTxK12rXHPKJl:b-i9s3yh) ([raw file](https://github.com/powershell/dscresources/blob/4b7a21c78536bf6af98ffa6667dca79d91646754/Naming.md#L13)):*
> high quality resource guidelines

maybe "High Quality DSC Resource Module" since that is the title of the document ('Creating a High Quality DSC Resource Module').

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresources/317)*
<!-- Sent from Reviewable.io -->

johlju commented:
My review comments (excepts for the typos ;) ) are meant for discussion.

PlagueHO commented:
It will be a bit difficult to determine if a completely community developed contributor module is actually HQRM (and remains HQRM over time). So I'm not sure how the naming standard "<name>Dsc" for HQRM could be enforced anyway. Unless the community is reviewing contributor modules it is only going to be a "good faith" effort around what modules meet HQRM or not. I personally have several modules that I believe meet HQRM standards, but as they've not undergone a full community review then it would be hard to say for certain.

I'm not actually sure what the best thing to do here is, but all of my partially reviewed modules have been upgraded to HQRM and were named as such. So I'd like to avoid going back and renaming them all back to 'c' again.

johlju commented:
I agree with @PlagueHO. The only time were we could try to enforce the HQRM guidelines, for a resource module a user/contributor owns, is if DSC Resource Kit (this repository) is the "umbrella" over them (but not supported by Microsoft!), linking community resource modules to/from this repo (still owned by the author/community), like with DscConfigurations repo. That would make the community resource modules that get's linked to get a sort of approval status, that they meet some critiera (HQRM, Support, Community etc).  
I would like to see @PlagueHO resource modules being linked to this repository, even if he still owns them. Having a resource module approved by the "DSC Resource Kit Community" would yield a higher value, it would show an interest by the community to support that module. If you understand what I'm getting at.
But I understand that it will come with administration that @zjalexander and @kwirkykat might not have time with.

If contributors/users community resource module should not be linked in anyway to this repo, then naming the modules, and HQRM is only relevant to the existing resources in DSC Resource Kit, which will then be Community Supported HQRM, but still under the umbrella of the DSC Resource Kit.

@PlagueHO Either case, you should not need to rename your new resource modules. I think your resource modules are a good example of a finished HQRM resource modules, ready to be supported by the community, and should be named with the suffix 'Dsc'. _You should definitely not rename them to 'c' as stated in this PR (Naming.md). :)_

zjalexander commented:
>I would like to see @PlagueHO resource modules being linked to this repository, even if he still owns them. Having a resource module approved by the "DSC Resource Kit Community" would yield a higher value, it would show an interest by the community to support that module

I'm totally open to this! The administration is definitely work and we would need to figure out a formal process for what that looks like. I'd love to keep that as community-centered as possible - maybe the community calls can play a role in that process?

PlagueHO commented:
Thanks for the support @johlju :grin: @zjalexander I'm up for being a trial run as a "community supported" resource (or whatever you're going to call it). I'd be happy to give this a go with [iSCSIDsc](https://github.com/PlagueHO/iSCSIDsc) (just need to change the license back to myself). Do you have a particular process you'd like me to follow in doing any of this?

zjalexander commented:
No, about 60% of the problem is we need to invent that process.

I'm out of office for labor day week (Sept 4-8) so I'll pick this up when I get back. I think we're getting somewhere though! :)

PlagueHO commented:
Cool as @zjalexander ! Have a great holiday!

FYI: I've reworked [iSCSIDsc](https://github.com/PlagueHO/iSCSIDsc) and released it now to the PSGallery. I've got 3 or 4 more resources to follow. So linking them to the DSC Resource kit somehow would be great, but no rush.

zjalexander commented:
Review status: 1 of 3 files reviewed at latest revision, 7 unresolved discussions.

---

*[CONTRIBUTING.md, line 367 at r2](https://reviewable.io:443/reviews/powershell/dscresources/317#-Ksigf_CbQN0VBVsnr-g:-KvnvkuUXDpqRIzHl8bU:b-896fix) ([raw file](https://github.com/powershell/dscresources/blob/4b7a21c78536bf6af98ffa6667dca79d91646754/CONTRIBUTING.md#L367)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Typo 'Any'
</blockquote></details>

Done.

---

*[CONTRIBUTING.md, line 397 at r2](https://reviewable.io:443/reviews/powershell/dscresources/317#-KsihwgRtN2YhPLrVcUu:-Kvnw5U5VZGVks9nfZCo:b-eiev1m) ([raw file](https://github.com/powershell/dscresources/blob/4b7a21c78536bf6af98ffa6667dca79d91646754/CONTRIBUTING.md#L397)):*
> All new resources in existing resource modules should
yeah, fixed that and removed the ambiguity

---

*[Maintainers.md, line 24 at r2](https://reviewable.io:443/reviews/powershell/dscresources/317#-KsiixuN6tqJ4lqqWh1N:-KvnwbPaqE1HoDqBZtf6:b-896fix) ([raw file](https://github.com/powershell/dscresources/blob/4b7a21c78536bf6af98ffa6667dca79d91646754/Maintainers.md#L24)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> up-to-date on  what

Two spaces between 'on' and 'what'.

</blockquote></details>

Done.

---

*[Maintainers.md, line 24 at r2](https://reviewable.io:443/reviews/powershell/dscresources/317#-Ksij4lNoN8JZucFHT6j:-KvnwbgB48r1QNTqg_2G:b-896fix) ([raw file](https://github.com/powershell/dscresources/blob/4b7a21c78536bf6af98ffa6667dca79d91646754/Maintainers.md#L24)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> repos

Change to repositories?
</blockquote></details>

Done.

---

*[Naming.md, line 12 at r2](https://reviewable.io:443/reviews/powershell/dscresources/317#-Ksik-zSA8lPhCxdIpSs:-KvnwpwupmyI83nC31d7:b-nndpj4) ([raw file](https://github.com/powershell/dscresources/blob/4b7a21c78536bf6af98ffa6667dca79d91646754/Naming.md#L12)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Is this a resource modules already present in the DSC Resource Kit only?  

When is it okay to rename a module from "x" to "Dsc"?  Community approval process? For example renaming xSQLServer. A community developed by a contributor outside of DSC Resource Kit we could only recommend not name to "Dsc" unless it's HQRM.
</blockquote></details>

I will clarify this in the doc

---

*[Naming.md, line 13 at r2](https://reviewable.io:443/reviews/powershell/dscresources/317#-KsijQahTxK12rXHPKJk:-Kvnx5YYbbzY24wSI2kC:b-896fix) ([raw file](https://github.com/powershell/dscresources/blob/4b7a21c78536bf6af98ffa6667dca79d91646754/Naming.md#L13)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> high quality resource guidelines

maybe "High Quality DSC Resource Module" since that is the title of the document ('Creating a High Quality DSC Resource Module').
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresources/317)*
<!-- Sent from Reviewable.io -->

zjalexander commented:
Review status: 0 of 3 files reviewed at latest revision, 7 unresolved discussions.

---

*[Maintainers.md, line 96 at r2](https://reviewable.io:443/reviews/powershell/dscresources/317#-Ksij9LRORxb0mIw_BlM:-KvnybEXHQXdfAk8ADZH:b-896fix) ([raw file](https://github.com/powershell/dscresources/blob/4b7a21c78536bf6af98ffa6667dca79d91646754/Maintainers.md#L96)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> reqeusts

Typo. And add full stop at the end.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresources/317)*
<!-- Sent from Reviewable.io -->

johlju commented:
Reviewed 3 of 3 files at r3.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[Naming.md, line 18 at r3](https://reviewable.io:443/reviews/powershell/dscresources/317#-Kvva3KqeuRgtm2_c47P:-Kvva3KrJyS2uXYd3H8d:b-nuo1vk) ([raw file](https://github.com/powershell/dscresources/blob/48f2cd9617dbd743ca38ecb1b5ed7e72edbc12b8/Naming.md#L18)):*
> submitted to the community call

Thinking we should add a text, or link to a text, how these should be submitted to be considered during the community call? 

Maybe adding a issue and prefixing the title with "Community Call:"? Users can't label there own issues, if they could we could use a label for that. Issues that should be brought up on the community call.

---

*[Naming.md, line 19 at r3](https://reviewable.io:443/reviews/powershell/dscresources/317#-Kvvb-2eskNAISIEWst6:-Kvvb-2eskNAISIEWst7:b-9ml2xi) ([raw file](https://github.com/powershell/dscresources/blob/48f2cd9617dbd743ca38ecb1b5ed7e72edbc12b8/Naming.md#L19)):*
> "A "HQRM" module follows the..."

Maybe "These modules should follow the..."

---

*[Naming.md, line 22 at r3](https://reviewable.io:443/reviews/powershell/dscresources/317#-KvvZxggI2nyA5OgdVGf:-KvvZxggI2nyA5OgdVGg:b-m2wrka) ([raw file](https://github.com/powershell/dscresources/blob/48f2cd9617dbd743ca38ecb1b5ed7e72edbc12b8/Naming.md#L22)):*
> ```Markdown
> This means the module has proper test coverage and follows Microsoft best practices for development of a DSC resource.
> Modules which have been reviewed by the community and meet the bar will be listed on this repository. 
> ```

Two blank lines here. Please remove one.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresources/317)*
<!-- Sent from Reviewable.io -->

johlju commented:
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

*[CONTRIBUTING.md, line 364 at r3](https://reviewable.io:443/reviews/powershell/dscresources/317#-Kw-mONb6WDu33fJMOsJ:-Kw-mONb6WDu33fJMOsK:b-jd9fwc) ([raw file](https://github.com/powershell/dscresources/blob/48f2cd9617dbd743ca38ecb1b5ed7e72edbc12b8/CONTRIBUTING.md#L364)):*
> If you are adding a new resource to an experimental/preview module (module name starts with 'x'), the resource name must also start with 'x' (e.g. MSFT_xResource.psm1 or xResource.psm1).


Couldn't we make the module name to start with 'x' to say it is unsupported, but the resource name don't need to start with 'x'?

That would help a scenario I discussed with @mgreenegit at MSIgnite (hope he correct me if I misunderstood something).
He was thinking of, emphasis on thinking of (!),  a third branch which could be, for example SqlServerDsc. While keeping xSQLServer-branch for experimental. 
dev would merge into xSQLServer-branch (master?) and maybe published to gallery by maintainers. And a third "supported"-branch which would be merged from xSQLServer-branch and that branch would be named SqlServerDsc and released (less frequently) by PowerShell DSC Team to the gallery. For this scenario it would be good if the resource names was not using 'x' in the names, just the module name should be using 'x' to indicate that the resource are not supported.

---

*[CONTRIBUTING.md, line 364 at r3](https://reviewable.io:443/reviews/powershell/dscresources/317#-Kw-mkWPR37Jm-basUA2:-Kw-mkWPR37Jm-basUA3:b-gk75gq) ([raw file](https://github.com/powershell/dscresources/blob/48f2cd9617dbd743ca38ecb1b5ed7e72edbc12b8/CONTRIBUTING.md#L364)):*
> ```Markdown
> Composite resource (with a configuration and a .psd1 file) files must have the exact same name as the resource or they will not be able to be imported. Hence, composite resource files should not have the MSFT_ prefix (e.g. xResource.psm1).
> 
> If you are adding a new resource to an experimental/preview module (module name starts with 'x'), the resource name must also start with 'x' (e.g. MSFT_xResource.psm1 or xResource.psm1).
> ```

Should say unsupported here.

---

*[Naming.md, line 10 at r3](https://reviewable.io:443/reviews/powershell/dscresources/317#-Kw-pIqJexB1VOqP0X3v:-Kw-pIqJexB1VOqP0X3w:bg12bbq) ([raw file](https://github.com/powershell/dscresources/blob/48f2cd9617dbd743ca38ecb1b5ed7e72edbc12b8/Naming.md#L10)):*
> ...means a resource...

Should be "...means a module..."? 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresources/317)*
<!-- Sent from Reviewable.io -->

zjalexander commented:
Review status: 1 of 3 files reviewed at latest revision, 6 unresolved discussions.

---

*[CONTRIBUTING.md, line 364 at r3](https://reviewable.io:443/reviews/powershell/dscresources/317#-Kw-mONb6WDu33fJMOsJ:-KwBG06-nnuwXzZILAKv:b-8cv5jn) ([raw file](https://github.com/powershell/dscresources/blob/48f2cd9617dbd743ca38ecb1b5ed7e72edbc12b8/CONTRIBUTING.md#L364)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> If you are adding a new resource to an experimental/preview module (module name starts with 'x'), the resource name must also start with 'x' (e.g. MSFT_xResource.psm1 or xResource.psm1).


Couldn't we make the module name to start with 'x' to say it is unsupported, but the resource name don't need to start with 'x'?

That would help a scenario I discussed with @mgreenegit at MSIgnite (hope he correct me if I misunderstood something).
He was thinking of, emphasis on thinking of (!),  a third branch which could be, for example SqlServerDsc. While keeping xSQLServer-branch for experimental. 
dev would merge into xSQLServer-branch (master?) and maybe published to gallery by maintainers. And a third "supported"-branch which would be merged from xSQLServer-branch and that branch would be named SqlServerDsc and released (less frequently) by PowerShell DSC Team to the gallery. For this scenario it would be good if the resource names was not using 'x' in the names, just the module name should be using 'x' to indicate that the resource are not supported.

</blockquote></details>

let me talk to Michael about this, I'm not sure I totally get it. 

---

*[CONTRIBUTING.md, line 364 at r3](https://reviewable.io:443/reviews/powershell/dscresources/317#-Kw-mkWPR37Jm-basUA2:-KwBG_uu6_3BAMYsLdu0:b-896fix) ([raw file](https://github.com/powershell/dscresources/blob/48f2cd9617dbd743ca38ecb1b5ed7e72edbc12b8/CONTRIBUTING.md#L364)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Should say unsupported here.
</blockquote></details>

Done.

---

*[Naming.md, line 10 at r3](https://reviewable.io:443/reviews/powershell/dscresources/317#-Kw-pIqJexB1VOqP0X3v:-KwBHfwia2ULEEZu_b6o:b-896fix) ([raw file](https://github.com/powershell/dscresources/blob/48f2cd9617dbd743ca38ecb1b5ed7e72edbc12b8/Naming.md#L10)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> ...means a resource...

Should be "...means a module..."? 
</blockquote></details>

Done.

---

*[Naming.md, line 18 at r3](https://reviewable.io:443/reviews/powershell/dscresources/317#-Kvva3KqeuRgtm2_c47P:-KwBIEaon5o4hWSZ5QTM:b-zgderm) ([raw file](https://github.com/powershell/dscresources/blob/48f2cd9617dbd743ca38ecb1b5ed7e72edbc12b8/Naming.md#L18)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> submitted to the community call

Thinking we should add a text, or link to a text, how these should be submitted to be considered during the community call? 

Maybe adding a issue and prefixing the title with "Community Call:"? Users can't label there own issues, if they could we could use a label for that. Issues that should be brought up on the community call.
</blockquote></details>

yeah, that makes sense. I'll add a link to CommunityAgenda.md, and after we finish this PR I'll take a crack at revising the community agenda to make it a bit more useful. 

---

*[Naming.md, line 19 at r3](https://reviewable.io:443/reviews/powershell/dscresources/317#-Kvvb-2eskNAISIEWst6:-KwBIVo3ASxhWd3tHUau:b-896fix) ([raw file](https://github.com/powershell/dscresources/blob/48f2cd9617dbd743ca38ecb1b5ed7e72edbc12b8/Naming.md#L19)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> "A "HQRM" module follows the..."

Maybe "These modules should follow the..."

</blockquote></details>

Done.

---

*[Naming.md, line 22 at r3](https://reviewable.io:443/reviews/powershell/dscresources/317#-KvvZxggI2nyA5OgdVGf:-KwBIZ6jLn3SEHeDdktK:b-896fix) ([raw file](https://github.com/powershell/dscresources/blob/48f2cd9617dbd743ca38ecb1b5ed7e72edbc12b8/Naming.md#L22)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Two blank lines here. Please remove one.
</blockquote></details>

Done.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresources/317)*
<!-- Sent from Reviewable.io -->

johlju commented:
Reviewed 2 of 2 files at r4.
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[CONTRIBUTING.md, line 364 at r3](https://reviewable.io:443/reviews/powershell/dscresources/317#-Kw-mONb6WDu33fJMOsJ:-KwBlCvVY0eA8ObL5TUK:bufwpqn) ([raw file](https://github.com/powershell/dscresources/blob/48f2cd9617dbd743ca38ecb1b5ed7e72edbc12b8/CONTRIBUTING.md#L364)):*
<details><summary><i>Previously, zjalexander (Zachary Alexander) wrote…</i></summary><blockquote>

let me talk to Michael about this, I'm not sure I totally get it. 
</blockquote></details>

Yeah, I think Michael can explain his thoughts better :smile: 

---

*[CONTRIBUTING.md, line 369 at r4](https://reviewable.io:443/reviews/powershell/dscresources/317#-KwBlQHhKXf8SoZdOrsz:-KwBlQHhKXf8SoZdOrt-:bwqvwkh) ([raw file](https://github.com/powershell/dscresources/blob/467821880072b1ec01c33d8a788bc08cd9daeb87/CONTRIBUTING.md#L369)):*
> ```Markdown
> Any test or example files for the resource should be named to match the files for the same resource. For example, if the main resource file is named 'MSFT_xResource.psm1', then the unit test file should be named 'MSFT_xResource.Tests.ps1'. Consistent naming helps the review process. 
> 
> For more details, please see [Naming](Naming.md)
> ```

Add full stop (.) at the end.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresources/317)*
<!-- Sent from Reviewable.io -->

tysonjhayes commented:
Reviewed 1 of 2 files at r4.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[Naming.md, line 14 at r4](https://reviewable.io:443/reviews/powershell/dscresources/317#-KwfZvCls4G7m5lhzqVR:-KwfZvCpflRknIKlts2y:btavuu) ([raw file](https://github.com/powershell/dscresources/blob/467821880072b1ec01c33d8a788bc08cd9daeb87/Naming.md#L14)):*
> ```Markdown
> Resource module authors should remove the "x" when they are ready to have a 1.0 version that is fit for public consumption.
> 
> ## Community Supported
> ```

Do we have a formalized list of what this is? I get its not the same list as the Microsoft Supported resources, but is it just everything under github.com/PowerShell that matches this naming convention?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresources/317)*
<!-- Sent from Reviewable.io -->

mgreenegit commented:
All, I have submitted a PR against this fork with significant changes to the guidelines for naming and support.  Could I please ask for community review?
https://github.com/zjalexander/DscResources/pull/2

johlju commented:
@mgreenegit Happy to review. Won’t be able to until Sunday or after the weekend though.

mgreenegit commented:
Here's a direct link to the Naming updates but please also review Supportability.md (linked from bottom of Naming.md page)
https://github.com/zjalexander/DscResources/blob/6d17a9af648fba61bbb81f9b339664db390b247c/Naming.md

PlagueHO commented:
Hi @mgreenegit - this is looking really good and clear.

One thing I'm not clear on is, should "Stable" versions of a resource be in a different GitHub repo from the "Active" version? It seems like the work "Branch" is being used to indicate something more akin to a "fork" rather than a branch? E.g. xPSDesiredStateConfiguration and PSDscResources are in separate repos.

I think maintaining separate repos for Active vs Stable releases might be the better way to go - rather than having a "Stable" branch in an "Active" repo. This is mainly because if you're looking for say NetworkingDsc and don't know about the "naming" convention you could easily get pretty confused.

Combining stable and active into the same repo also gets a bit more problematic for repos with Wikis (it is still manageable though).

Or am I just getting confused by use of the term "branch"?

But over all, this is looking to clear a lot of things up! Thank you @mgreenegit and @zjalexander

mgreenegit commented:
@PlagueHO @smurawski @zjalexander @kwirkykat Regarding the previous comment on branches, I would like to consider, if we are making this level of change, shifting to eliminate the "x" entirely in favor of just the DSC suffix and semver.  Feedback?

johlju commented:
Reviewed 2 of 2 files at r5.
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---

*[Naming.md, line 17 at r5](https://reviewable.io:443/reviews/powershell/dscresources/317#-Kx8ukL1iVOlaiNGi_eP:-Kx8ukL1iVOlaiNGi_eQ:bxrw4m9) ([raw file](https://github.com/powershell/dscresources/blob/9762e3f4083c9b3c6529c31dc0cab6ef3128c89a/Naming.md#L17)):*
> ```Markdown
> and Microsoft, accountable to quality through a process that includes:
> - Code review prior to listing as submodule in [DSC Resource](http://github.com/powershell/dscresources) repo
>     - Includes an expectation of [project structure](Contributing.md##developing-a-new-resource), including [documentation](Contributing.md##writing-documentation), [tests](Contributing.md###write-tests) and [CI](Contributing.md###tests-in-appveyor) process to revalidate at every push to source
> ```

These links seems to be broken. Please see review comment under the section Active.

---

*[Naming.md, line 24 at r5](https://reviewable.io:443/reviews/powershell/dscresources/317#-Kx8ofF91l2yPHmkaAj6:-Kx8ofF91l2yPHmkaAj7:b914hdg) ([raw file](https://github.com/powershell/dscresources/blob/9762e3f4083c9b3c6529c31dc0cab6ef3128c89a/Naming.md#L24)):*
> Issues

Should we have lower case  'i' on 'issues'?

---

*[Naming.md, line 31 at r5](https://reviewable.io:443/reviews/powershell/dscresources/317#-Kx8qDf9v6iMuk0Pt8bk:-Kx8qDf9v6iMuk0Pt8bl:bk708gh) ([raw file](https://github.com/powershell/dscresources/blob/9762e3f4083c9b3c6529c31dc0cab6ef3128c89a/Naming.md#L31)):*
> ```Markdown
> The "x" prefix designates a branch of a project that is under active development.
> Frequent releases are to be expected
> and major releases that include [breaking changes](Contributing.md##breaking-changes)
> ```

I think it should only be one '#' for this link to work.

Also, I can only get the link working if i use the CONTRIBUTING.md with capitalized letters (the correct filename of the file)... might be my browser? 

---

*[Supportability.md, line 83 at r5](https://reviewable.io:443/reviews/powershell/dscresources/317#-Kx8wjyP6XEXVI6ReWmg:-Kx8wjyP6XEXVI6ReWmh:b4unoa) ([raw file](https://github.com/powershell/dscresources/blob/9762e3f4083c9b3c6529c31dc0cab6ef3128c89a/Supportability.md#L83)):*
> ```Markdown
> when managed using PowerShell cmdlets (in the same security context as DSC).
> 
> Together, the teams work with the customer to resolve the issue.
> ```

Maybe mention that if the customer needs to submit an issue to the 'community maintained Open Source solution' the issue might not be resolved until the community resolves the issue? Or, is the Microsoft Support issue considered resolved when the issue is submitted to the 'community maintained Open Source solution'? If so, mention that?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresources/317)*
<!-- Sent from Reviewable.io -->

johlju commented:
I'm in favor of eliminating the 'x' in favor of SemVer and using 'Dsc' suffix, for all resources. I think it will make it easier to use milestones. Easier to target breaking changes to a particular milestone. Potentially there could be "active" branches connected to major milestones (that is not released, or released as `x.y.z-alpha.1`). Such milestone branch could be merged into the "main active"-branch and released as `x.y.z-rc.1`.

tysonjhayes commented:
Ditch the x! It's served it's purpose.

---

Review status: all files reviewed at latest revision, 7 unresolved discussions.

---

*[Supportability.md, line 83 at r5](https://reviewable.io:443/reviews/powershell/dscresources/317#-Kx8wjyP6XEXVI6ReWmg:-KxA3E6ZeCLlRHVWx9i4:brhspg8) ([raw file](https://github.com/powershell/dscresources/blob/9762e3f4083c9b3c6529c31dc0cab6ef3128c89a/Supportability.md#L83)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Maybe mention that if the customer needs to submit an issue to the 'community maintained Open Source solution' the issue might not be resolved until the community resolves the issue? Or, is the Microsoft Support issue considered resolved when the issue is submitted to the 'community maintained Open Source solution'? If so, mention that?
</blockquote></details>

Yeah it kind of sounds like in this scenario this is a problem in the xActiveDirectory resource, in which case isn't it up to the community to fix as MS doesn't really support this? I think calling this out is useful. Sets the bar around support. 

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresources/317#-:-KxA31fK-WYfQIvak-Qb:bc9n5q8)*
<!-- Sent from Reviewable.io -->

smurawski commented:
I gotta say, I'm totally confused by the guidance around naming.

We've got `x` modules and resources that are supposed to be under "active development" and not production ready, but can be "high quality".

However, the bar for becoming "stable" is "production ready" in some places in the doc, but in others also requires "high quality".

The problem I see is that we are trying to rationalize the use of "x" which existed before these modules were open sourced.  It's time to eat a bit of pain and ditch that prefixing.

The suffixing of modules with a conflicting in-box module seems reasonable to me, however as a global pattern I don't like it.  What if my module contains both functions and resources (which is what I would recommend - the resources should be a thin layer around the functions in the module)?  Do I need to suffix it with DSC just because there are resources there?  Isn't that kind of the point of some of the metadata that PowerShellGet offers?

It seems like we are trying to re-solve problems with naming conventions (x/stable should just be handled by following semver and module name conflicts should be handled as an exception rather than the rule).  This adds a ton of complexity for the end user of resources.  You'll end up with two versions of resources and modules with completely different versioning.

I think we need to step back and re-visit some CD practices.  We have one branch and that is master.  We use semver to indicate the level of change in each release.  We focus on unit and integration tests that validate the behavior of the resources inside the modules.

I love the effort everyone here is putting into making a strong community of resources around DSC.  I just  cannot see participating myself as things are so convoluted.  Naming, versioning, and release patterns just seem inscrutable to me.  

In the Chef world (which isn't perfect nor the right way to do everything, but is a major point of reference for me in OSS process), I've got no problem contributing to/maintaining/releasing cookbooks.  There is an understanding about the behavior.  Releases usually follow semver, with major releases noting breaking changes, minor releases indicating new features or substantial updates, and patch releases indicating minor bug fixes.  Users of these control the level of risk they want to take by being as aggressive or lax in version pinning in their environments.

My 2 cents, for what it is worth, is drop the "x" and use "DSC" suffixing as only a way to deal with in-box module conflict (or 3rd party module naming conflict).  Don't invest in complicated branching process - master should be the only long-lived branch.  Others should be short-lived until their PR is merged.  Make it simple - modules with DSC resources are just PowerShell modules at the end of the day.

zjalexander commented:
ok, I hope we addressed feedback from the call, @smurawski , and @kwirkykat . Take another look when you can

johlju commented:
This text looks much better - cleaner! Great work! :smile: 

---

Reviewed 1 of 2 files at r6, 1 of 1 files at r7.
Review status: all files reviewed at latest revision, 14 unresolved discussions.

---

*[Naming.md, line 18 at r7](https://reviewable.io:443/reviews/powershell/dscresources/317#-Kyw3YxrbQVA70k9IpDM:-Kyw3YxrbQVA70k9IpDN:bmxo2js) ([raw file](https://github.com/powershell/dscresources/blob/d9c96770dd216a8f3cb7f71458525c55004b474f/Naming.md#L18)):*
> repo

Maybe write out to 'repository'?

---

*[Naming.md, line 25 at r7](https://reviewable.io:443/reviews/powershell/dscresources/317#-Kyw3fKI0Mgfwc8jy8yI:-Kyw3fKI0Mgfwc8jy8yJ:b-2k5hz3) ([raw file](https://github.com/powershell/dscresources/blob/d9c96770dd216a8f3cb7f71458525c55004b474f/Naming.md#L25)):*
> Issues

Maybe 'issues' (lower-case 'i')

---

*[Naming.md, line 38 at r7](https://reviewable.io:443/reviews/powershell/dscresources/317#-Kyw54GRx1BH-0fDSq8Q:-Kyw54GRx1BH-0fDSq8R:bb0x3fc) ([raw file](https://github.com/powershell/dscresources/blob/d9c96770dd216a8f3cb7f71458525c55004b474f/Naming.md#L38)):*
> "DSC Resource"

Is it 'DSC Resource' and not 'DSCResource'?

The .psd1 look like this in xSQLServer.
```
 Tags = @('DesiredStateConfiguration', 'DSC', 'DSCResourceKit', 'DSCResource')
```

---

*[Naming.md, line 39 at r7](https://reviewable.io:443/reviews/powershell/dscresources/317#-Kyw47ynzGk83tOcgg5g:-Kyw47ynzGk83tOcgg5h:b-vv2k9f) ([raw file](https://github.com/powershell/dscresources/blob/d9c96770dd216a8f3cb7f71458525c55004b474f/Naming.md#L39)):*
> [gallery]

Maybe '[PowerShell Gallery]'?

---

*[Naming.md, line 48 at r7](https://reviewable.io:443/reviews/powershell/dscresources/317#-Kyw63QaHjZ43I-wXcHW:-Kyw63QaHjZ43I-wXcHX:b-u12awj) ([raw file](https://github.com/powershell/dscresources/blob/d9c96770dd216a8f3cb7f71458525c55004b474f/Naming.md#L48)):*
> "DSC Resource"

See previous comment about this.

---

*[Naming.md, line 49 at r7](https://reviewable.io:443/reviews/powershell/dscresources/317#-Kyw67cdRWcXHtIbtZZY:-Kyw67cdRWcXHtIbtZZZ:b-vv2k9f) ([raw file](https://github.com/powershell/dscresources/blob/d9c96770dd216a8f3cb7f71458525c55004b474f/Naming.md#L49)):*
> [gallery]

Maybe '[PowerShell Gallery]'?

---

*[Naming.md, line 62 at r7](https://reviewable.io:443/reviews/powershell/dscresources/317#-Kyw6XjUtZUTTDqFUvWx:-Kyw6XjUtZUTTDqFUvWy:bb6yr5b) ([raw file](https://github.com/powershell/dscresources/blob/d9c96770dd216a8f3cb7f71458525c55004b474f/Naming.md#L62)):*
> repo

Maybe move 'repo' out from the link text, keeping just 'DscResources' (lower-case 'sc'), and write it out as 'repository'?

---

*[Naming.md, line 79 at r7](https://reviewable.io:443/reviews/powershell/dscresources/317#-Kyw7OMjUI7pB3aQcYHR:-Kyw7OMjUI7pB3aQcYHS:b-vv2k9f) ([raw file](https://github.com/powershell/dscresources/blob/d9c96770dd216a8f3cb7f71458525c55004b474f/Naming.md#L79)):*
> [gallery]

Maybe '[PowerShell Gallery]'?

---

*[Supportability.md, line 32 at r7](https://reviewable.io:443/reviews/powershell/dscresources/317#-Kyw8dyPyqxfX5RuHyPu:-Kyw8dyPyqxfX5RuHyPv:b-m8cpp3) ([raw file](https://github.com/powershell/dscresources/blob/d9c96770dd216a8f3cb7f71458525c55004b474f/Supportability.md#L32)):*
> Resources

Maybe 'resources' (lower-case 'r')?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresources/317#-:-Kyw9M4E13EFNvbaMPxH:b-9nj7q)*
<!-- Sent from Reviewable.io -->

mgreenegit commented:
Review status: all files reviewed at latest revision, 14 unresolved discussions.

---

*[CONTRIBUTING.md, line 369 at r4](https://reviewable.io:443/reviews/powershell/dscresources/317#-KwBlQHhKXf8SoZdOrsz:-Kz-cUSTM9v3klGw0fqx:bb6un1j) ([raw file](https://github.com/powershell/dscresources/blob/467821880072b1ec01c33d8a788bc08cd9daeb87/CONTRIBUTING.md#L369)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Add full stop (.) at the end.
</blockquote></details>

Done

---

*[Naming.md, line 14 at r4](https://reviewable.io:443/reviews/powershell/dscresources/317#-KwfZvCls4G7m5lhzqVR:-Kz-w5U7_47D21Fk9Fs-:bqov080) ([raw file](https://github.com/powershell/dscresources/blob/467821880072b1ec01c33d8a788bc08cd9daeb87/Naming.md#L14)):*
<details><summary><i>Previously, tysonjhayes (Tyson J. Hayes) wrote…</i></summary><blockquote>

Do we have a formalized list of what this is? I get its not the same list as the Microsoft Supported resources, but is it just everything under github.com/PowerShell that matches this naming convention?
</blockquote></details>

the discussion on the community call landed with it is up to the author to make a decision.  we might need to iterate on guidance in the future.

---

*[Naming.md, line 17 at r5](https://reviewable.io:443/reviews/powershell/dscresources/317#-Kx8ukL1iVOlaiNGi_eP:-Kz-vXgNEWuGn3wFgcKs:blngdbr) ([raw file](https://github.com/powershell/dscresources/blob/9762e3f4083c9b3c6529c31dc0cab6ef3128c89a/Naming.md#L17)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

These links seems to be broken. Please see review comment under the section Active.
</blockquote></details>

fixed.  GH url's are case sensitive.

---

*[Naming.md, line 31 at r5](https://reviewable.io:443/reviews/powershell/dscresources/317#-Kx8qDf9v6iMuk0Pt8bk:-Kz-vqDeWehChO5Gs09_:bb7f2mf) ([raw file](https://github.com/powershell/dscresources/blob/9762e3f4083c9b3c6529c31dc0cab6ef3128c89a/Naming.md#L31)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

I think it should only be one '#' for this link to work.

Also, I can only get the link working if i use the CONTRIBUTING.md with capitalized letters (the correct filename of the file)... might be my browser? 
</blockquote></details>

done

---

*[Naming.md, line 18 at r7](https://reviewable.io:443/reviews/powershell/dscresources/317#-Kyw3YxrbQVA70k9IpDM:-Kz-e8d2qhYk-oiLBq-G:bb6un1j) ([raw file](https://github.com/powershell/dscresources/blob/d9c96770dd216a8f3cb7f71458525c55004b474f/Naming.md#L18)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> repo

Maybe write out to 'repository'?
</blockquote></details>

Done

---

*[Naming.md, line 25 at r7](https://reviewable.io:443/reviews/powershell/dscresources/317#-Kyw3fKI0Mgfwc8jy8yI:-Kz-vdRcAOx-Mn5UBX6Y:bb7f2mf) ([raw file](https://github.com/powershell/dscresources/blob/d9c96770dd216a8f3cb7f71458525c55004b474f/Naming.md#L25)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> Issues

Maybe 'issues' (lower-case 'i')
</blockquote></details>

done

---

*[Naming.md, line 38 at r7](https://reviewable.io:443/reviews/powershell/dscresources/317#-Kyw54GRx1BH-0fDSq8Q:-Kz-vtTbYDjGB2aqbSjQ:bb7f2mf) ([raw file](https://github.com/powershell/dscresources/blob/d9c96770dd216a8f3cb7f71458525c55004b474f/Naming.md#L38)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> "DSC Resource"

Is it 'DSC Resource' and not 'DSCResource'?

The .psd1 look like this in xSQLServer.
```
 Tags = @('DesiredStateConfiguration', 'DSC', 'DSCResourceKit', 'DSCResource')
```
</blockquote></details>

done

---

*[Naming.md, line 39 at r7](https://reviewable.io:443/reviews/powershell/dscresources/317#-Kyw47ynzGk83tOcgg5g:-Kz-vxdXsHFKbyR7GxgU:bb7f2mf) ([raw file](https://github.com/powershell/dscresources/blob/d9c96770dd216a8f3cb7f71458525c55004b474f/Naming.md#L39)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> [gallery]

Maybe '[PowerShell Gallery]'?
</blockquote></details>

done

---

*[Naming.md, line 48 at r7](https://reviewable.io:443/reviews/powershell/dscresources/317#-Kyw63QaHjZ43I-wXcHW:-Kz-wDwXzc7OxbPQDmsD:bb7f2mf) ([raw file](https://github.com/powershell/dscresources/blob/d9c96770dd216a8f3cb7f71458525c55004b474f/Naming.md#L48)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> "DSC Resource"

See previous comment about this.
</blockquote></details>

done

---

*[Naming.md, line 49 at r7](https://reviewable.io:443/reviews/powershell/dscresources/317#-Kyw67cdRWcXHtIbtZZY:-Kz-wEsHV51NUhtmbEVD:bb7f2mf) ([raw file](https://github.com/powershell/dscresources/blob/d9c96770dd216a8f3cb7f71458525c55004b474f/Naming.md#L49)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> [gallery]

Maybe '[PowerShell Gallery]'?
</blockquote></details>

done

---

*[Naming.md, line 62 at r7](https://reviewable.io:443/reviews/powershell/dscresources/317#-Kyw6XjUtZUTTDqFUvWx:-Kz-wMmDAm4Js4K98O7k:bb7f2mf) ([raw file](https://github.com/powershell/dscresources/blob/d9c96770dd216a8f3cb7f71458525c55004b474f/Naming.md#L62)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> repo

Maybe move 'repo' out from the link text, keeping just 'DscResources' (lower-case 'sc'), and write it out as 'repository'?
</blockquote></details>

done

---

*[Naming.md, line 79 at r7](https://reviewable.io:443/reviews/powershell/dscresources/317#-Kyw7OMjUI7pB3aQcYHR:-Kz-wSk8aKFm0BVpWTOV:bb7f2mf) ([raw file](https://github.com/powershell/dscresources/blob/d9c96770dd216a8f3cb7f71458525c55004b474f/Naming.md#L79)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> [gallery]

Maybe '[PowerShell Gallery]'?
</blockquote></details>

done

---

*[Supportability.md, line 83 at r5](https://reviewable.io:443/reviews/powershell/dscresources/317#-Kx8wjyP6XEXVI6ReWmg:-Kz-x5LGbnvQdN4Bcx1p:b-l284lv) ([raw file](https://github.com/powershell/dscresources/blob/9762e3f4083c9b3c6529c31dc0cab6ef3128c89a/Supportability.md#L83)):*
<details><summary><i>Previously, tysonjhayes (Tyson J. Hayes) wrote…</i></summary><blockquote>

Yeah it kind of sounds like in this scenario this is a problem in the xActiveDirectory resource, in which case isn't it up to the community to fix as MS doesn't really support this? I think calling this out is useful. Sets the bar around support. 
</blockquote></details>

Good advice.  I added a new sentence but not sure if it goes far enough.

---

*[Supportability.md, line 32 at r7](https://reviewable.io:443/reviews/powershell/dscresources/317#-Kyw8dyPyqxfX5RuHyPu:-Kz-xNHrglE7ixG4ZRxs:bocr982) ([raw file](https://github.com/powershell/dscresources/blob/d9c96770dd216a8f3cb7f71458525c55004b474f/Supportability.md#L32)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

> Resources

Maybe 'resources' (lower-case 'r')?
</blockquote></details>

I was on the fence about whether resources is a proper noun.  Switched all instances to lower case instead.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresources/317)*
<!-- Sent from Reviewable.io -->

johlju commented:
@mgreenegit I don't think you pushed the changes? I can't see any changes were done on those review comments you wrote done on.

mgreenegit commented:
It's a PR against the repo for this PR, and hasn't been merged yet:
https://github.com/zjalexander/DscResources/commit/818fc95a3ecdcafc04bfbc95dc9112a4494cc622
@zjalexander will have to accept first

zjalexander commented:
@mgreenegit its merged now :)

johlju commented:
@zjalexander @mgreenegit the PR was merged into the master branch in your fork, and not the update-guidelines branch. That's why the changes has not shown up here. :)

mgreenegit commented:
The correct PR is pending
https://github.com/zjalexander/DscResources/pull/5
@zjalexander

johlju commented:
For some reason a few conversations did not want to resolve when I acknowledge it. Maybe needs a 'Done'.

---

Reviewed 3 of 3 files at r8.
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

*[CONTRIBUTING.md, line 369 at r4](https://reviewable.io:443/reviews/powershell/dscresources/317#-KwBlQHhKXf8SoZdOrsz:-Kz-z5WEDSWxaSNsdGU3:b-ir5hrl) ([raw file](https://github.com/powershell/dscresources/blob/467821880072b1ec01c33d8a788bc08cd9daeb87/CONTRIBUTING.md#L369)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary><blockquote>

Done
</blockquote></details>

Missed this one I think.

---

*[Naming.md, line 49 at r7](https://reviewable.io:443/reviews/powershell/dscresources/317#-Kyw67cdRWcXHtIbtZZY:-L-CzSncu_abI5skPaqz:b6n0751) ([raw file](https://github.com/powershell/dscresources/blob/d9c96770dd216a8f3cb7f71458525c55004b474f/Naming.md#L49)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary><blockquote>

done
</blockquote></details>

This one was missed.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresources/317#-:-L-D-jxfyUv_FRqOD38Z:b-gghwzy)*
<!-- Sent from Reviewable.io -->

mgreenegit commented:
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

*[CONTRIBUTING.md, line 369 at r4](https://reviewable.io:443/reviews/powershell/dscresources/317#-KwBlQHhKXf8SoZdOrsz:-L-Hr0fqo6SWsO5N4uML:bb7f2mf) ([raw file](https://github.com/powershell/dscresources/blob/467821880072b1ec01c33d8a788bc08cd9daeb87/CONTRIBUTING.md#L369)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

Missed this one I think.
</blockquote></details>

done

---

*[Naming.md, line 49 at r7](https://reviewable.io:443/reviews/powershell/dscresources/317#-Kyw67cdRWcXHtIbtZZY:-L-HquW4T7Udq7zKqDSR:bb7f2mf) ([raw file](https://github.com/powershell/dscresources/blob/d9c96770dd216a8f3cb7f71458525c55004b474f/Naming.md#L49)):*
<details><summary><i>Previously, johlju (Johan Ljunggren) wrote…</i></summary><blockquote>

This one was missed.
</blockquote></details>

done

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresources/317)*
<!-- Sent from Reviewable.io -->

mgreenegit commented:
New PR pending @zjalexander 
https://github.com/zjalexander/DscResources/pull/6

johlju commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 2 of 2 files at r9.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresources/317#-:-L-It0UDl0grHBb6TPkM:bnfp4nl)*
<!-- Sent from Reviewable.io -->

mgreenegit commented:
Reviewed 1 of 3 files at r3, 1 of 3 files at r8, 2 of 2 files at r9.
Review status: all files reviewed at latest revision, 3 unresolved discussions.

---

*[Naming.md, line 14 at r4](https://reviewable.io:443/reviews/powershell/dscresources/317#-KwfZvCls4G7m5lhzqVR:-L-XWeugwObiE4-maNN4:b2ni9fl) ([raw file](https://github.com/powershell/dscresources/blob/467821880072b1ec01c33d8a788bc08cd9daeb87/Naming.md#L14)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary><blockquote>

the discussion on the community call landed with it is up to the author to make a decision.  we might need to iterate on guidance in the future.
</blockquote></details>

one

---

*[Naming.md, line 17 at r5](https://reviewable.io:443/reviews/powershell/dscresources/317#-Kx8ukL1iVOlaiNGi_eP:-L-XWdsqVU5XV8Y2rzty:bb7f2mf) ([raw file](https://github.com/powershell/dscresources/blob/9762e3f4083c9b3c6529c31dc0cab6ef3128c89a/Naming.md#L17)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary><blockquote>

fixed.  GH url's are case sensitive.
</blockquote></details>

done

---

*[Supportability.md, line 83 at r5](https://reviewable.io:443/reviews/powershell/dscresources/317#-Kx8wjyP6XEXVI6ReWmg:-L-XWbr9JbXczRZBI7H6:bb7f2mf) ([raw file](https://github.com/powershell/dscresources/blob/9762e3f4083c9b3c6529c31dc0cab6ef3128c89a/Supportability.md#L83)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary><blockquote>

Good advice.  I added a new sentence but not sure if it goes far enough.
</blockquote></details>

done

---

*[Supportability.md, line 32 at r7](https://reviewable.io:443/reviews/powershell/dscresources/317#-Kyw8dyPyqxfX5RuHyPu:-L-XWa6NIwiDnk8ayDv0:bb7f2mf) ([raw file](https://github.com/powershell/dscresources/blob/d9c96770dd216a8f3cb7f71458525c55004b474f/Supportability.md#L32)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary><blockquote>

I was on the fence about whether resources is a proper noun.  Switched all instances to lower case instead.
</blockquote></details>

done

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresources/317)*
<!-- Sent from Reviewable.io -->

mgreenegit commented:
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[Naming.md, line 14 at r4](https://reviewable.io:443/reviews/powershell/dscresources/317#-KwfZvCls4G7m5lhzqVR:-L-XWo50HmQ8Kf20isXe:b32w0x) ([raw file](https://github.com/powershell/dscresources/blob/467821880072b1ec01c33d8a788bc08cd9daeb87/Naming.md#L14)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary><blockquote>

one
</blockquote></details>

OK

---

*[Supportability.md, line 83 at r5](https://reviewable.io:443/reviews/powershell/dscresources/317#-Kx8wjyP6XEXVI6ReWmg:-L-XWirRWXrL9e88ZMZr:b32wtd) ([raw file](https://github.com/powershell/dscresources/blob/9762e3f4083c9b3c6529c31dc0cab6ef3128c89a/Supportability.md#L83)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary><blockquote>

done
</blockquote></details>

ok

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresources/317)*
<!-- Sent from Reviewable.io -->

mgreenegit commented:
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[Naming.md, line 14 at r4](https://reviewable.io:443/reviews/powershell/dscresources/317#-KwfZvCls4G7m5lhzqVR:-L-XWwzwLNlII7yWcqvh:bb7f2mf) ([raw file](https://github.com/powershell/dscresources/blob/467821880072b1ec01c33d8a788bc08cd9daeb87/Naming.md#L14)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary><blockquote>

OK
</blockquote></details>

done

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresources/317)*
<!-- Sent from Reviewable.io -->

mgreenegit commented:
Review status: all files reviewed at latest revision, 2 unresolved discussions.

---

*[Supportability.md, line 83 at r5](https://reviewable.io:443/reviews/powershell/dscresources/317#-Kx8wjyP6XEXVI6ReWmg:-L-XX1jBYLaTeA1hbnEH:bb7f2mf) ([raw file](https://github.com/powershell/dscresources/blob/9762e3f4083c9b3c6529c31dc0cab6ef3128c89a/Supportability.md#L83)):*
<details><summary><i>Previously, mgreenegit (Michael Greene) wrote…</i></summary><blockquote>

ok
</blockquote></details>

done

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/powershell/dscresources/317)*
<!-- Sent from Reviewable.io -->

mgreenegit commented:
@kwirkykat ready to merge

brwilkinson commented:
Please Rfc tag these kinds of changes if possible. /cc @zjalexander
